packages/core/src/Stack/Stack.js

```diff
@@ -0,0 +1,155 @@
+import React, { useMemo, useCallback, useRef } from "react";
+import PropTypes from "prop-types";
+import clsx from "clsx";
+import { withStyles, Divider, useTheme } from "@material-ui/core";
+import isString from "lodash/isString";
+import isBoolean from "lodash/isBoolean";
+import styles from "./styles";
+import useWidth from "../utils/useWidth";
+import Focus from "../Focus";
+
+/**
+ * A Stack component allows the organization of its children in a vertical or horizontal layout.
+ *
+ * It also allows the specification of the spacing between the stack elements and the adition of a divider between the elements.
+ */
+const HvStack = (props) => {
+  const {
+    classes,
+    children,
+    direction = "column",
+    spacing = "sm",
+    divider = false,
+    withNavigation = false,
+    dividerProps = {},
+    ...others
+  } = props;
+  const width = useWidth();
+  const theme = useTheme();
+  const containerRef = useRef(null);
+
+  const widths = useMemo(() => [...theme.breakpoints.keys], [theme]);
+
+  /**
+   * @returns {string} - Returns a direction for the stack: column or row. If the
+   *                     `direction` property is a string and a valid direction then we
+   *                     use it. If it's an object with multiple directions by breakpoint
+   *                     we use the appropriate one or search for the nearest breakpoint
+   *                     smaller than the current one to use.
+   */
+  const getDirection = useCallback(() => {
```

Comment:
This function doesn't receive any parameter, yet in line 52 some parameters are passed. However receiving the parameters may be good, if it receives parameters and avoids accessing the values from context the function could be pure and avoid the `useCallback`